iceoryx_hoofs/include/iceoryx_hoofs/memory/relative_pointer.inl

```diff
@@ -239,11 +230,71 @@ inline typename RelativePointer<T>::ptr_t RelativePointer<T>::computeRawPtr() co
     return getPtr(segment_id_t{m_id}, m_offset);
 }
 
+// AXIVION Next Construct AutosarC++19_03-A15.5.3, AutosarC++19_03-A15.4.2, FaultDetection-NoexceptViolations : False
+// positive, std::terminate is not called in the c'tor of PointerRepository and noexcept-specification is not violated
 inline PointerRepository<segment_id_underlying_t, UntypedRelativePointer::ptr_t>& getRepository() noexcept
 {
+    // AXIVION Next Construct AutosarC++19_03-A3.3.2 : PointerRepository can't be constexpr, usage of the static
+    // object is encapsulated in the RelativePointer class
     static PointerRepository<segment_id_underlying_t, UntypedRelativePointer::ptr_t> repository;
     return repository;
 }
+
+template <typename T>
+// AXIVION Next Line AutosarC++19_03-A13.5.5 : The RelativePointer shall explicitly be comparable to raw pointers
```

Comment:
Why does static analysis have a problem with that? It is a legitimate use case and no problem at all?

Especially since it just reads values and compares them without changing the underlying objects. To make this clear we need `const T* rhs` though. Maybe this would remove the Axivion warning as well.

Reply:
Quote from the AUTOSAR document:

```
Comparison operators shall be non-member functions with identical parameter types and noexcept.
```

Reply:
@MatthiasKillat I think in general for let's say higher-level application development it's an ok-ish rule. However, you are right that for an base library it's not very adequate.

Reply:
The main question for me is how often we need to suppress this rule. If it is only in the lower single digit order then I wouldn't care but if it is in the double digit order then it would be more honest to just disable it.

Reply:
Okay, it's not even in the lower single digit in this file :sweat_smile:

Reply:
In any case I do not think it is a good rule, regardless how often it applies.